### PR TITLE
Configure NI name whenever configuring type

### PR DESCRIPTION
**Pull request by kjahed**
Leaf ref validation on the router side requires list keys to be configured. This PR changes all tests that configure the default NI _type_ to also configure it's _name_. Note that we use update instead of replace to avoid deleting any configuration under the NI done outside the test (e..g, static routes).

### CODE REVIEW

**Comment by Ankur19** [comment]
This looks like a very large scale change.

Can you instead write a method in fptest (https://github.com/openconfig/featureprofiles/blob/main/internal/fptest/networkinstance.go) say `ConfigureDefaultNetworkInstance()` and then update all instances to use it?

File: feature/networkinstance/otg_tests/defaults_test/defaults_test.go [hunk 1]
```diff
@@ -160,11 +160,15 @@ func TestDefaultAddressFamilies(t *testing.T) {
...
 				fptest.SetPortSpeed(t, dutP2)
 			}
+
+			d := &oc.Root{}
 			if tc.niName == deviations.DefaultNetworkInstance(dut) {
+				ni := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+				ni.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE
 				dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
-				gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+				gnmi.Update(t, dut, dutConfNIPath.Config(), ni)
```
Comment (Ankur19) [inline feature/networkinstance/otg_tests/defaults_test/defaults_test.go:169]: `fptest.ConfigureDefaultNetworkInstance(t, dut)`

File: feature/networkinstance/ate_tests/defaults_test/defaults_test.go [hunk 2]
```diff
@@ -157,12 +157,15 @@ func TestDefaultAddressFamilies(t *testing.T) {
 				fptest.SetPortSpeed(t, dutP1)
 				fptest.SetPortSpeed(t, dutP2)
 			}
+
+			d := &oc.Root{}
 			if tc.niName == deviations.DefaultNetworkInstance(dut) {
+				ni := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
```
Comment (Ankur19) [inline feature/networkinstance/ate_tests/defaults_test/defaults_test.go:163]: `fptest.ConfigureDefaultNetworkInstance(t, dut)`

File: feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go [hunk 3]
```diff
@@ -143,7 +144,10 @@ func configureDUT(t *testing.T, name string, lldpEnabled bool) (*ondatra.DUTDevi
 	gnmi.Replace(t, node, lldp.Enabled().Config(), lldpEnabled)
 
 	if lldpEnabled {
-		gnmi.Replace(t, node, lldp.Interface(p.Name()).Enabled().Config(), lldpEnabled)
+		gnmi.Replace(t, node, lldp.Interface(p.Name()).Config(), &oc.Lldp_Interface{
+			Name:    ygot.String(p.Name()),
```
Comment (Ankur19) [inline feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go:148]: This looks unrelated to NI

File: feature/lldp/ate_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go [hunk 4]
```diff
@@ -99,7 +99,10 @@ func configureNode(t *testing.T, name string, lldpEnabled bool) (*ondatra.DUTDev
 	gnmi.Replace(t, node, lldp.Enabled().Config(), lldpEnabled)
 
 	if lldpEnabled {
-		gnmi.Replace(t, node, lldp.Interface(p.Name()).Enabled().Config(), lldpEnabled)
+		gnmi.Replace(t, node, lldp.Interface(p.Name()).Config(), &oc.Lldp_Interface{
```
Comment (Ankur19) [inline feature/lldp/ate_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go:102]: Unrelated to NI

File: feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go [hunk 5]
```diff
@@ -751,7 +751,10 @@ func TestP4rtNodeID(t *testing.T) {
 				t.Fatalf("Couldn't find P4RT Node for port: %s", "port1")
 			}
 			t.Logf("Configuring P4RT Node: %s", nodes["port1"])
-			gnmi.Replace(t, dut, gnmi.OC().Component(nodes["port1"]).IntegratedCircuit().Config(), ic)
+			gnmi.Replace(t, dut, gnmi.OC().Component(nodes["port1"]).Config(), &oc.Component{
```
Comment (Ankur19) [inline feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:754]: Unrelated to NI

File: feature/experimental/system/gnmi/benchmarking/ate_tests/full_configuration_replace_test/full_configuration_replace_test.go [hunk 6]
```diff
@@ -61,13 +61,12 @@ func TestGnmiFullConfigReplace(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
 
 	t.Log("Configure network instance on DUT")
-	dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
-	gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	fptest.ConfigureDefaultNetworkInstance(t, dut)
 
 	t.Log("Cleanup exisitng BGP and ISIS configs on DUT before configuring test configs")
-	dutBGPPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	dutBGPPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
```
Comment (Ankur19) [inline feature/experimental/system/gnmi/benchmarking/ate_tests/full_configuration_replace_test/full_configuration_replace_test.go:67]: Unrelated to NI

File: feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go (hunk 3 above)
Comment (kjahed) [reply feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go:148]: Yes, but this a similar problem that needs to be fixed. Interface name must be configured before configuring enabled. Below are similar. Should I submit these changes in separate PRs?

Comment (Ankur19) [reply feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go:148]: Yes please. Thank you for the contribution.

File: feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go (hunk 5 above)
Comment (kjahed) [reply feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go]: Removed

File: feature/lldp/ate_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go (hunk 4 above)
Comment (kjahed) [reply feature/lldp/ate_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go]: Removed

File: feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go (hunk 3 above)
Comment (kjahed) [reply feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go]: Done

File: feature/networkinstance/ate_tests/defaults_test/defaults_test.go (hunk 2 above)
Comment (kjahed) [reply feature/networkinstance/ate_tests/defaults_test/defaults_test.go]: Done

File: feature/networkinstance/otg_tests/defaults_test/defaults_test.go (hunk 1 above)
Comment (kjahed) [reply feature/networkinstance/otg_tests/defaults_test/defaults_test.go]: Done

**Review by guoshiuan — APPROVED** [review]
